src/reporting/ReportingWindow.sol

```diff
@@ -0,0 +1,289 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/factories/MarketFactory.sol';
+import 'ROOT/factories/SetFactory.sol';
+import 'ROOT/factories/RegistrationTokenFactory.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/RegistrationToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+
+
+contract ReportingWindow is DelegationTarget, Typed, Initializable {
+    Branch private branch;
+    uint256 private startTime;
+    RegistrationToken private registrationToken;
+    Set private markets;
+    Set private limitedReporterMarkets;
+    Set private allReporterMarkets;
+    mapping(address => Set) private reportsByReporter;
+    mapping(address => uint256) private numberOfReportsByMarket;
+    uint256 private constant REPORTING_DURATION_SECONDS = 27 * 1 days;
+    uint256 private constant REPORTING_DISPUTE_DURATION_SECONDS = 3 days;
+    uint256 private constant BASE_MINIMUM_REPORTERS_PER_MARKET = 7;
+
+    function initialize(Branch _branch, uint256 _reportingWindowId) public beforeInitialized returns (bool) {
+        endInitialization();
+        branch = _branch;
+        startTime = _reportingWindowId * branch.getReportingPeriodDurationInSeconds();
+        RegistrationTokenFactory _registrationTokenFactory = RegistrationTokenFactory(controller.lookup("RegistrationTokenFactory"));
+        registrationToken = _registrationTokenFactory.createRegistrationToken(controller, this);
+        SetFactory _setFactory = SetFactory(controller.lookup("SetFactory"));
+        markets = _setFactory.createSet(controller, this);
+        limitedReporterMarkets = _setFactory.createSet(controller, this);
+        allReporterMarkets = _setFactory.createSet(controller, this);
+        return true;
+    }
+
+    function createNewMarket(uint256 _endTime, int256 _numOutcomes, int256 _payoutDenominator, int256 _feePerEthInWei, address _denominationToken, address _creator, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, int256 _topic) public payable returns (IMarket _newMarket) {
```

Comment:
Though, if changing that ends up being a big change (shouldn't be, but might) we can put that change off until after the Solidity migration is complete.